projects/origami/index.js

```diff
@@ -13,18 +13,51 @@ module.exports = {
 Object.keys(GRAPH_URLS).forEach(chain => {
   const endpoint = GRAPH_URLS[chain]
   module.exports[chain] = {
-    tvl: async (api) => {
-      const { metrics: [{ investmentVaults }] } = await cachedGraphQuery('origami/' + chain, endpoint, '{ metrics { investmentVaults { id } } }')
-      const vaults = investmentVaults.map(vault => vault.id)
-      let tokens = await api.multiCall({  abi: 'address:reserveToken', calls: vaults })
-      if (chain === 'arbitrum')
-        tokens = await api.multiCall({  abi: 'address:baseToken', calls: tokens })
-
-      const decimals = await api.multiCall({  abi: 'uint8:decimals', calls: vaults })
-      const supplies = await api.multiCall({  abi: 'uint256:totalSupply', calls: vaults })
-      const reserves = await api.multiCall({  abi: 'uint256:reservesPerShare', calls: vaults })
-      const bals = reserves.map((reserve, i) => reserve * supplies[i] / 10 ** decimals[i])
-      api.add(tokens, bals)
-    }
+    tvl: (api) => tvl(api, chain, endpoint),
+    borrowed: (api) => liabilities(api, chain, endpoint)
   }
 })
+
+async function tvl(
+  api,
+  chain,
+  endpoint,
+) {
+  const { metrics: [{  investmentVaults }] } = await cachedGraphQuery('origami/' + chain, endpoint, '{ metrics { investmentVaults { id kinds } } }')
+  await Promise.all(
+    investmentVaults.map(async vault => {
+      const isLeveraged = !!vault.kinds.find( v => v === 'Leverage')
+      let token = await api.call({ abi: 'address:reserveToken', target: vault.id })
+
+      if (isLeveraged) {
+        const assetsAndLiabilities = await api.call({ abi: 'function assetsAndLiabilities() external view returns (uint256 assets,uint256 liabilities,uint256 ratio)', target: vault.id })
```

Comment:
These rpc calls are being done twice in both `tvl` and `liabilities` functions, would it be possible to make the rpc calls before hand and pass the results as params of each function? tried couple of different things but couldn't make it work

Reply:
hi @frost-ostrich thanks for the update, I refactored it to batch query. Also, removed the borrowed field as we use it only for lending protocols not leverage protocols (atm)

Reply:
@g1nt0ki origami is similar to Contango in that regard, which does show borrowed 

https://defillama.com/protocol/contango-v2#information

Could you please advise why it does show the debt but origami cannot? 

Thank you!

Reply:
@g1nt0ki could you please check the above comment when you get time?